### PR TITLE
refaactor(unlock-app): splitting the confirmation screen

### PR DESCRIPTION
**Pull request by julien51**
# Description

Our confirmation screen was getting seriously busy with lots of stuff in it.
I am refactoring by splitting it in multiple "flavors".

There is certainly more DRY-ing up to do but I think that is a first approach.
I want to test thoroughly each of the flow:
* credit card purchase
* claim
* swap and purchase
* purchase

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCard.tsx [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    switch (error.code) {
+      case -32000:
+      case 4001:
+      case 'ACTION_REJECTED':
+        ToastHelper.error('Transaction rejected.')
+        break
+      default:
+        ToastHelper.error(message || error?.error?.message || error.message)
+    }
+  }
```
Comment (searchableguy) [inline unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCard.tsx:205]: I think you can move this outside code and use it as default error handler.

File: unlock-app/src/components/interface/checkout/main/Confirm/ConfirmClaim.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    switch (error.code) {
+      case -32000:
+      case 4001:
+      case 'ACTION_REJECTED':
+        ToastHelper.error('Transaction rejected.')
+        break
+      default:
+        ToastHelper.error(message || error?.error?.message || error.message)
+    }
+  }
```
Comment (searchableguy) [inline unlock-app/src/components/interface/checkout/main/Confirm/ConfirmClaim.tsx:121]: Same. Repetition.

File: unlock-app/src/components/interface/checkout/main/checkoutMachine.ts [hunk 3]
```diff
@@ -154,7 +154,7 @@ type Payment =
     }
   | {
       method: 'swap_and_purchase'
-      route: any
+      route?: object
```
Comment (searchableguy) [inline unlock-app/src/components/interface/checkout/main/checkoutMachine.ts:157]:
```suggestion
      route?: any
```

I will keep any here or find a route type from uniswap sdk.

**Review by searchableguy — COMMENTED** [review]
I think we can dry the code a lot here but let's at least do the easy bits.

File: unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCard.tsx (hunk 1 above)
Comment (julien51) [reply unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCard.tsx:205]: yeah good point. Let me clean it up.

**Review by searchableguy — APPROVED** [review]
LGTM. One question regarding not handling transaction rejection errors.

File: unlock-app/src/components/interface/checkout/main/Confirm.tsx [hunk 4]
```diff
@@ -1,734 +1,90 @@
...
-      case 4001:
-      case 'ACTION_REJECTED':
-        ToastHelper.error('Transaction rejected.')
-        break
-      default:
-        ToastHelper.error(message || error?.error?.message || error.message)
-    }
+  const onError = (message: string) => {
+    ToastHelper.error(message)
   }
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/components/interface/checkout/main/Confirm.tsx:32]: yes we are, there: https://github.com/unlock-protocol/unlock/pull/11898/files#diff-9986e83341da673bd65505c350028f3a3a15ffb84708b21b9ac48c72cfd5ea36R213

**Pull request merged**